test/test_distributed_hydrostatic_model.jl

```diff
@@ -78,49 +78,55 @@ Ny = 32
-        v = Array(interior(v))
-        w = Array(interior(w))
-        c = Array(interior(c))
-        η = Array(interior(η))
-
-        cpu_arch = cpu_architecture(arch)
-
-        us = partition_global_array(cpu_arch, us, size(u))
-        vs = partition_global_array(cpu_arch, vs, size(v))
-        ws = partition_global_array(cpu_arch, ws, size(w))
-        cs = partition_global_array(cpu_arch, cs, size(c))
-        ηs = partition_global_array(cpu_arch, ηs, size(η))
-
-        atol = eps(eltype(grid))
-        rtol = sqrt(eps(eltype(grid)))
-
-        @test all(isapprox(u, us; atol, rtol))
-        @test all(isapprox(v, vs; atol, rtol))
-        @test all(isapprox(w, ws; atol, rtol))
-        @test all(isapprox(c, cs; atol, rtol))
-        @test all(isapprox(η, ηs; atol, rtol))
+        underlying_grid = LatitudeLongitudeGrid(arch, size = (Nx, Ny, 1),
+                                                halo = (3, 3, 3),
+                                                latitude = (-80, 80),
+                                                longitude = (-160, 160),
+                                                z = (-1, 0),
+                                                radius = 1,
+                                                topology=(Bounded, Bounded, Bounded))
+
+        bottom(λ, φ) = -30 < λ < 30 && -40 < φ < 20 ? 0 : - 1
+
+        immersed_grid = ImmersedBoundaryGrid(underlying_grid, GridFittedBottom(bottom); active_cells_map = true)
+
+        for grid in (underlying_grid, immersed_grid)
+            global_grid = reconstruct_global_grid(grid)
+
+            # "s" for "serial" computation
+            us, vs, ws, cs, ηs = solid_body_rotation_test(global_grid)
+
+            us = Array(interior(us))
+            vs = Array(interior(vs))
+            ws = Array(interior(ws))
+            cs = Array(interior(cs))
+            ηs = Array(interior(ηs))
+
+            @info "  Testing distributed solid body rotation with architecture $arch on $(typeof(grid).name.wrapper)"
+            u, v, w, c, η = solid_body_rotation_test(grid)
+
+            u = Array(interior(u))
+            v = Array(interior(v))
+            w = Array(interior(w))
+            c = Array(interior(c))
+            η = Array(interior(η))
+
+            cpu_arch = cpu_architecture(arch)
+
+            us = partition_global_array(cpu_arch, us, size(u))
+            vs = partition_global_array(cpu_arch, vs, size(v))
+            ws = partition_global_array(cpu_arch, ws, size(w))
+            cs = partition_global_array(cpu_arch, cs, size(c))
+            ηs = partition_global_array(cpu_arch, ηs, size(η))
+
+            atol = eps(eltype(grid))
+            rtol = sqrt(eps(eltype(grid)))
+
+            @test all(isapprox(u, us; atol, rtol))
+            @test all(isapprox(v, vs; atol, rtol))
+            @test all(isapprox(w, ws; atol, rtol))
+            @test all(isapprox(c, cs; atol, rtol))
+            @test all(isapprox(η, ηs; atol, rtol))
```

Comment:
Is is possible to write this test to work with `Field` and `on_architecture` and avoid using `Array`?

Reply:
first, we need an `on_architecture` method for fields. I was thinking already about it but I think it should be done in another PR.

Reply:
Let's wait for #3490 to be merged then?